pkg/sql/logictest/testdata/logic_test/strict_ddl_atomicity

```diff
@@ -0,0 +1,40 @@
+# Default behavior, where atomicity violations are allowed
+
+statement ok
+CREATE TABLE testing (k int, v string);
+  INSERT INTO testing (k,v) VALUES (1, 'a'), (2, 'b'), (3, 'a'), (4, 'b')
+
+statement ok
+BEGIN
+
+statement ok
+ALTER TABLE testing ADD CONSTRAINT "unique_values" UNIQUE(v)
+
+statement ok
+INSERT INTO testing (k,v) VALUES (5, 'c')
```

Comment:
The test does not reflect what I was running into. The `INSERT` was not on `testing` and was not dependent in any way on the new constraint. Instead, it was on a completely separate table and was **not** rolled back.

Reply:
OK I will add the other thing you say, but note that in this test here the insert is not rolled back either.